### PR TITLE
Updating developer instructions

### PR DESCRIPTION
**Pull request by dpgraham**
* Amended `npm run shrinkwrap-prod` to include `npm install --no-shrinkwrap` which re-installs dev dependencies without updating shrinkwrap
* Changed release guide to not include the instruction to re-install dev dependencies, `npm run shrinkwrap-prod` covers this now
* Moved the site doc update documentation into a separate section

### CODE REVIEW

File: docs/en/contributing-to-appium/developers-overview.md [hunk 1]
```diff
@@ -139,6 +139,22 @@ _FORCE_LOGS=1 <command> # show module log output during test run
 Note that we have a convention for unit test files to end in `-specs.js` and
 for e2e test files to end in `-e2e-specs.js`.
 
+### Updating Appium site docs
+
+The [Appium site](https://appium.io) is built using a site generator that translates the markdown in https://github.com/appium/appium to static HTML
+
+To update the site docs, follow these steps:
+
+1 From Appium repo, run `npm run generate-docs` and make a pull request if there are any newly generated docs.
```
Comment (jlipps) [inline docs/en/contributing-to-appium/developers-overview.md]: don't we need `.` after the numerals?

File: docs/en/contributing-to-appium/developers-overview.md [hunk 2]
```diff
@@ -180,21 +196,12 @@ converted into the `npm-shrinkwrap.json` file.
 1. Determine whether we have a `patch` (bugfix), `minor` (feature), or `major` (breaking) release according to the principles of SemVer.
 1. Update `package.json` with the appropriate new version.
 1. Update the CHANGELOG/README with appropriate changes and submit for review as a PR, along with shrinkwrap and `package.json` changes. Wait for it to be merged, then pull it into the release branch.
-1. Run `npm run prod-shrinkwrap` to install just the production dependencies and create a shrinkwrap that only locks production dependencies.
+1. Run `npm run shrinkwrap-prod`. This script removes dev dependencies (leaving only  production dependencies), creates a production-only `npm-shrinkwrap.json` and then re-installs the dev dependencies by doing `npm install --no-shrinkwrap`.
```
Comment (jlipps) [inline docs/en/contributing-to-appium/developers-overview.md]: extra space in here

File: docs/en/contributing-to-appium/developers-overview.md [hunk 3]
```diff
@@ -180,21 +196,12 @@ converted into the `npm-shrinkwrap.json` file.
...
 1. Create a tag of the form `v<version>` on the release branch (usually a minor branch like `1.5` or `1.4`), with: `git tag -a v<version>`, e.g., `git tag -a v1.5.0`. This is not necessary for beta versions.
 1. Push the tag to upstream: `git push --tags <remote> <branch>`
-1. Install dev dependencies (or at least `gulp` and `appium-gulp-plugins`), and undo the changes to the NPM shrinkwrap JSON file (e.g., `git checkout -- npm-shrinkwrap.json`).
 1. Run `npm publish` (with `--tag beta` if this isn't an official release).
 1. Remove the NPM shrinkwrap JSON file from Git and push the changes
 1. Update the site docs or put in an [issue](https://github.com/appium/appium/issues) to have the site docs updated
```
Comment (jlipps) [inline docs/en/contributing-to-appium/developers-overview.md]: maybe should add (`see section above on updating site docs`)?

**Review by jlipps — CHANGES_REQUESTED** [review]

**Comment by dpgraham** [comment]
Fixes pushed.

File: docs/en/contributing-to-appium/developers-overview.md [hunk 4]
```diff
@@ -133,12 +133,28 @@ npm run lint            # run eslint on the code
 npm run watch           # watch directory to re-transpile on code change, lint, and run unit tests
 npm run test            # same as above but don't watch
 npm run e2e-test        # transpile and run end-to-end/functional tests
-_FORCE_LOGS=1 <command> # show module log output during test run
+_FORCE_LOGS=1. <command> # show module log output during test run
```
Comment (imurchie) [inline docs/en/contributing-to-appium/developers-overview.md]: What is the period for here?

File: docs/en/contributing-to-appium/developers-overview.md [hunk 5]
```diff
@@ -133,12 +133,28 @@ npm run lint            # run eslint on the code
...
+The [Appium site](https://appium.io) is built using a site generator that translates the markdown in https://github.com/appium/appium to static HTML
+
+To update the site docs, follow these steps:
+
+1. From Appium repo, run `npm run generate-docs` and make a pull request if there are any newly generated docs.
+1. When the above PR is merged, clone the [appium.io](https://github.com/appium/appium.io) repo (if you haven't done so already)
+1. Check out and pull the latest `gh-pages` branch
+1. Follow the setup guide in the README.md
+1. Generate the docs by running `npm run build:docs`
+1. Test by running `npm run serve` and navigating to https://localhost:4000/
```
Comment (imurchie) [inline docs/en/contributing-to-appium/developers-overview.md]: What does "Test" mean in this context? Are there steps that ought to be gone through to make sure that things are right?

Comment (dpgraham) [reply docs/en/contributing-to-appium/developers-overview.md]: I could probably just take this out. I don't think the site needs to be tested, the CI handles that well enough.

File: docs/en/contributing-to-appium/developers-overview.md (hunk 4 above)
Comment (dpgraham) [reply docs/en/contributing-to-appium/developers-overview.md]: That's a typo

**Pull request merged**